### ISSUE TITLE
[Feature] Progressive Web App (PWA) Support 

### ISSUE THREAD

**Issue by Ronald-Diemicke** [rommapp/romm#1027]
**Is your feature request related to a problem? Please describe.**
I'd like to be able to play my library from my phone while offline and then synchronize my progress back to the server when I've got a connection.

**Describe the solution you'd like**
Progressive Web Apps (PWAs) are a way to get a web app into a storefront OR install a web app to a device. Once the app is installable as a PWA you can use Service Workers to run background tasks (IE like metadata updates or synchronizing save states)

Additionally - this would be nice if you could use the browser cache (https://developer.mozilla.org/en-US/docs/Web/API/Cache) or IndexedDB (https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API) to save roms into storage when offline.

One other idea in the same vein would be if you could also connect the client to multiple servers (ala Plex) this would also make this so you could make this more flexible.

**Describe alternatives you've considered**
The only other option for this would be a fully native mobile app for Android & iPhone and to have it handshake with the Romm server(s)

**Comment by Casuallynoted** [comment]
If I recall correctly, Romm did have PWA support at one point but the login system was kind of weird about it and threw a lot of CSRF errors so it got disabled. I remember some discussion about switching the login system to something that wouldn't have that issue so I don't see why this wouldn't be possible once something like that is done. I'm not an expert on all this though so I'm gonna wait for one of the devs to confirm ^^;

**Comment by TheInfamousAlk** [comment]
Still looking forward to this. I play half on my PC and half on my bedroom TV by way of my phone, streamed to the chromecast. Kind of roundabout, but it's the best way I can find to put romm on my tv via chromecast.

**Comment by gantoine** [comment]
Closing this in favour of future improvement to the android and ios apps.
